src/www/model/server.ts

```diff
@@ -14,13 +14,28 @@
 
 // TODO: add guidelines for this file
 
+export enum ServerType {
+  // The connection data is static, doesn't change, and isn't deleted on disconnect.
+  STATIC_CONNECTION,
+
+  // The connection data is refetched via the access key on each connection.
+  // and deleted on each disconnection
+  DYNAMIC_CONNECTION,
+}
+
 export interface Server {
   // A unique id that identifies this Server.
   readonly id: string;
 
+  // A type specifying the manner in which the Server connects.
+  readonly type: ServerType;
+
   // The name of this server, as given by the user.
   name: string;
```

Comment:
The consumer of `Server` shouldn't be able to change the name, sessionConfigLocation and address. Those are all controlled by the server implementation. They should probably be getters instead. But it's fine to keep it as is for this PR for now.

Reply:
Yeah, agreed. Didn't want to get too much into those weeds with that, we've gone over enough in this PR.

Reply:
Could you add a TODO so we don't forget?